nixnet/types.py

```diff
@@ -361,14 +371,27 @@ def __repr__(self):
         # type: () -> typing.Text
         """CanFrame debug representation.
 
-        >>> CanFrame(1, constants.FrameType.CAN_DATA)
-        CanFrame(CanIdentifier(0x1), echo=False, type=FrameType.CAN_DATA, timestamp=0x0, payload=...)
+        >>> CanFrame(1)
+        CanFrame(CanIdentifier(0x1))
+        >>> CanFrame(1, constants.FrameType.CANFD_DATA, b'\x01')
+        CanFrame(CanIdentifier(0x1), type=FrameType.CANFD_DATA, len(payload)=1)
         """
-        return "CanFrame({}, echo={}, type={}, timestamp=0x{:x}, payload=...)".format(
+        optional = []
+        if self.echo:
+            optional.append('echo={}'.format(self.echo))
+        if self.type != constants.FrameType.CAN_DATA:
+            optional.append('type={}'.format(self.type))
+        if self.timestamp != 0:
+            optional.append('timestamp=0x{:x}'.format(self.timestamp))
+        if self.payload:
+            optional.append('len(payload)={}'.format(len(self.payload)))
+        if optional:
+            optional_params = ', {}'.format(", ".join(optional))
+        else:
+            optional_params = ''
+        return "CanFrame({}{})".format(
             self.identifier,
-            self.echo,
-            self.type,
-            self.timestamp)
+            optional_params)
```

Comment:
I like it.

I'm not certain why we need the optional flags.  My initial thought was to print all of them.  But, then I read the [definition of repr: ](https://docs.python.org/2/library/repr.html)
class repr.Repr

    ...size limits for different object types are added to avoid the generation of representations which are excessively long.

You're right about length.  So, I'm okay with the optional flags if you guys want them.